libraries/browser-tracker-core/src/helpers/cross_domain.ts

```diff
@@ -0,0 +1,55 @@
+import { ExtendedCrossDomainLinker, ExtendedCrossDomainLinkerAttributes } from '../tracker/types';
+
+const DEFAULT_CROSS_DOMAIN_LINKER_PARAMS: ExtendedCrossDomainLinkerAttributes = {
+  sessionId: true,
+  sourceId: true,
+  sourcePlatform: false,
+  userId: false,
+  reason: false,
+};
+
+export function createCrossDomainParameterValue(
+  isExtendedFormat: boolean,
+  attributeConfiguration: ExtendedCrossDomainLinkerAttributes | undefined,
+  attributeValues: ExtendedCrossDomainLinker & {
+    /* As `reason` might be a callback, we also need to pass the event to calculate the reason value. */
+    event: Event;
+  }
+): string {
+  let crossDomainParameterValue;
+  const timestamp = new Date().getTime();
+  const config = { ...DEFAULT_CROSS_DOMAIN_LINKER_PARAMS, ...attributeConfiguration };
+  const { domainUserId, userId, sessionId, sourceId, sourcePlatform, event } = attributeValues;
+
+  const eventTarget = event.currentTarget as HTMLAnchorElement | HTMLAreaElement | null;
+  const reason = typeof config.reason === 'function' ? config.reason(event) : eventTarget?.textContent?.trim();
+
+  if (isExtendedFormat) {
+    /* Index is used by Enrich, so it should not be changed. */
+    crossDomainParameterValue = [
+      domainUserId,
+      timestamp,
+      config.sessionId && sessionId,
+      config.userId && urlSafeBase64Encode(userId || ''),
+      config.sourceId && sourceId,
```

Comment:
I don't think there are any restrictions on the format of appId; using a domain (that will contain `.`) is not unheard of. You may need to b64 this, too.

Reply:
Will do, thanks